forge/test/operators/pytorch/nn/test_softmax.py

```diff
@@ -0,0 +1,436 @@
+
+# SPDX-License-Identifier: Apache-2.0
+
+# Tests for testing of nn operators
+#
+# In this test we use pytorch tensors and operators to verify forge operators
+
+
+# GENERAL OP SUPPORT TEST PLAN:
+# 1. Operand type - any supported type
+# 2. Operand source(s):
+# (+)  2.1 From another op
+#       - Operator -> input
+# (+)  2.2 From DRAM queue
+#       - input_queue flag = false
+#       - Special case of From host? May it be triggered if the operator is not the first node of the network?
+#       - Can this be triggered from pybuda.Parameter?
+#       - Can this be triggered from big pybuda.Constant?
+# (+)  2.3 Const Inputs (const eval pass)
+#       - Operator where all inputs are constants. Does it make difference if tensor is big > L1
+#       - Verification via netlists that scenario is triggered???
+# (+)  2.4 From host
+#       - Input tensor as input of network -> Operator is first node in network and input_queue flag = true
+#       - Can this scenario be triggered from pybuda.Parameter?
+#       - Can this be triggered from big pybuda.Constant?
+# 3 Operand shapes type(s):
+# (+)  3.1 Full tensor (i.e. full expected shape)
+#       - Is 3 dims max for all ops? Ex. Conv is 3d max
+# (+)  3.2 Tensor reduce on one or more dims to 1
+#       - Vector
+#       - Only one dim is not equal to 1
+# (+)  3.3 Scalar
+#       - Create tensor of dimension equal to 0 (tensor from scalar) or just to use scalar as simple value
+# 4. Operand / output size of dimensions (few examples of each, 10 values total)
+# (+)  4.1 Divisible by 32
+# (+)  4.2 Prime numbers
+# (+)  4.3 Very large (thousands, 10s of thousands)
+#       - 100x100, 100x1000
+#       - maybe nightly only
+# (+)  4.4 Extreme ratios between height/width
+#      4.5 ...probably many more interesting combinations here
+# 5. Data format - all supported formats
+# (/)  5.1 Output DF
+# (/)  5.2 Intermediate DF
+# (/)  5.3 Accumulation DF
+# (+)  5.4 Operand DFs
+# (+) 6. Math fidelity - LoFi, HiFi2a, Hifi2b, Hifi3, Hifi4
+# (/) 7. Special attributes - if applicable.. like approx_mode for Exp, for example
+
+
+import pytest
+import torch
+
+from typing import List, Dict, Type
+from loguru import logger
+
+import forge
+import forge.op
+
+from forge.op_repo import TensorShape
+
+from test.operators.utils import InputSourceFlags, VerifyUtils
+from test.operators.utils import ShapeUtils
+from test.operators.utils import FailingReasons
+from test.operators.utils.utils import TestDevice
+from test.operators.utils import PytestParamsUtils
+from test.operators.utils import ValueRanges
+
+
+class ModelFromAnotherOp(torch.nn.Module):
+
+    model_name = "model_op_src_from_another_op"
+
+    def __init__(self, shape, dim):
+        super(ModelFromAnotherOp, self).__init__()
+        self.testname = "Softmax_operator_test_op_src_from_another_op"
+        self.shape = shape
+        self.dim = dim
+        self.softmax = torch.nn.Softmax(dim=self.dim)
```

Comment:
@nvukobratTT 
Here we use class variant of softmax https://pytorch.org/docs/stable/generated/torch.nn.Softmax.html. There is also functional variant https://pytorch.org/docs/stable/generated/torch.nn.functional.softmax.html.
Class variant is basically wrapper around functional so it probably doesn't make difference which we select.
Maybe package `nn` is too wide here. I suggest we should something more specific like use `activations`.

Reply:
I don't have hard preferences on this one. Using either class definition or functional one would output the same results compiler wise :)) 

From a model perspective, it's common to see both of these, so let's choose one of the approaches for our testing, and if needed we can extend it for other definitions as well (but this is unlikely)

Reply:
I agree that nn package is too wide to organize our tests in that way. We should consider to reorganize tests in another PR.